app/controllers/users_controller.rb

```diff
@@ -39,6 +39,7 @@ def user_params
     params
       .require(:user)
       .permit(:name, :email, :password, :mobile_number, :role)
+      .transform_values(&:presence)
```

Comment:
I initially thought I could use `compact_blank` here, but this removes the key/value pair instead of just setting the value to nil. This could lead to scenarios such as:
1. Admin opens edit user form.
2. Admin deletes email address and presses save.
3. Message "User successfully updated" is shown, when in fact the form should be re-rendered with an error so that the admin knows this action is not permitted.